perun-core/src/main/java/cz/metacentrum/perun/core/impl/ExtSourceISXML.java

```diff
@@ -0,0 +1,103 @@
+
+	@Override
+	protected InputStream createTwoWaySSLConnection(String uri) throws IOException, InternalErrorException {
+		//prepare sslFactory
+		SSLSocketFactory factory = (SSLSocketFactory) SSLSocketFactory.getDefault();
+		HttpsURLConnection.setDefaultSSLSocketFactory(factory);
+
+		URL myurl = new URL(uri);
+		this.setCon((HttpURLConnection) myurl.openConnection());
+
+		// Prepare the basic auth, if the username and password was specified
+		if (getAttributes().get("user") != null && getAttributes().get("password") != null) {
+			String val = (new StringBuffer(getAttributes().get("user")).append(":").append(getAttributes().get("password"))).toString();
+
+			Base64 encoder = new Base64();
+			String base64Encoded = new String(encoder.encode(val.getBytes()));
+			base64Encoded = base64Encoded.trim();
+			String authorizationString = "Basic " + base64Encoded;
+			getCon().setRequestProperty("Authorization", authorizationString);
+		}
+		getCon().setAllowUserInteraction(false);
+
+		//prepare query for IS MU page
+		ismuquery = getQueryForGroup();
+
+		//set request header if is required (set in extSource xml)
+		this.getCon().setDoOutput(true);
+		this.getCon().setRequestProperty("Content-Type", "multipart/form-data; boundary=" + boundary);
+
+		try (
+			OutputStream output = getCon().getOutputStream();
+			PrintWriter writer = new PrintWriter(new OutputStreamWriter(output, StandardCharsets.UTF_8), true);
+		) {
+			// Send param about return
+			writer.append("--" + boundary).append(CRLF);
+			writer.append("Content-Disposition: form-data; name=\"out\"").append(CRLF);
+			writer.append(CRLF).append("xml").append(CRLF).flush();
+
+			// Send xml file.
+			writer.append("--" + boundary).append(CRLF);
+			writer.append("Content-Disposition: form-data; name=\"xml\"; filename=\"perun.xml\"").append(CRLF);
+			writer.append("Content-Type: text/xml; charset=" + StandardCharsets.UTF_8).append(CRLF); // Text file itself must be saved in this charset!
+			writer.append(CRLF).flush();
+			writer.append(ismuquery);
+			output.flush(); // Important before continuing with writer!
+			writer.append(CRLF).flush(); // CRLF is important! It indicates end of boundary.
+
+			// End of multipart/form-data.
+			writer.append("--" + boundary + "--").append(CRLF).flush();
+		}
+
+		//String text = "";
+		int responseCode = this.getCon().getResponseCode();
+		if(responseCode == 200) {
+			InputStream is = this.getCon().getInputStream();
+			return is;
+		}
+
+		throw new InternalErrorException("Wrong response code while opening connection on uri '" + uri + "'. Response code: " + responseCode);
+	}
+
+	/**
+	 * Prepare request for getting group information.
+	 *
+	 * @return request with specific settings for group (by XML setting in perun-extSource.xml file)
+	 */
+	private String getQueryForGroup() {
+		return	"<?xml version=\"1.0\" encoding=\"utf-8\"?>\n" +
+				"<request>\n" +
+				"<skupina_clenove reqid=\"5\">\n" +
```

Comment:
Please generate a random reqid code here and log it into the Perun logs. If there will be an error on IS side, they will tag that error with the reqid.